### PR TITLE
Fix/stoploss on exchange dryrun

### LINKED ISSUE #2785: trailing_stop = True not selling?
## Step 1: Have you search for this issue before posting it?
Yes

## Step 2: Describe your environment

  * Operating system: ubuntu server
  * Python Version: 3.8
  * CCXT version: ccxt==1.21.56
  * Branch: Develop
  * Last Commit ID: 2f82122fc40caf428d48f7ab942c8dc49c3ee8f1
  
## Step 3: Describe the problem:
trailing stop-loss not selling

Should it not have sold the trade already?
Trade ID: 1 (since 7 hours ago)
Current Pair: EOS/BTC
Amount: 14.59 (0.006 BTC)
Open Rate: 0.00041110
Current Rate: **0.00042120**
Current Profit: 2.25%
Initial Stoploss: 0.00040987 (-0.30%)
Stoploss: **0.00042472** (-0.30%)

Trade ID: 2 (since 6 hours ago)
Current Pair: BNB/BTC
Amount: 3.02 (0.006 BTC)
Open Rate: 0.00198260
Current Rate: **0.00200150**
Current Profit: 0.75%
Initial Stoploss: 0.00197665 (-0.30%)
Stoploss: **0.00200178** (-0.30%)

Trade ID: 3 (since 4 hours ago)
Current Pair: DASH/BTC
Amount: 0.504 (0.006 BTC)
Open Rate: 0.01188600
Current Rate: **0.01288400**
Current Profit: 8.18%
Initial Stoploss: 0.01185034 (-0.30%)
Stoploss: **0.01327705** (-0.30%)

The price was above the stop-loss at one point but it didn't sell. I'm running in dry-mode so it should not be a problem that the sell did not occur.
**Config:**
stoploss = -0.003

# Trailing stoploss
trailing_stop = True
trailing_only_offset_is_reached = False
trailing_stop_positive = 0.003
#trailing_stop_positive_offset = 0.20  # Disabled / not configure

### Steps to reproduce:
Not sure :-(
  
### Observed Results:
It just keeps on not selling.

### Relevant code exceptions or logs:

  ```
  // paste your log here
  ```

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
stoploss_on_exchange in combination with dry-run needs to emulate using a regular stoploss.

strangely enough, handling for this case is in place when determining the sell-rate - but not to trigger the stoploss sell itself - making the part below dead code without this new addition.

https://github.com/freqtrade/freqtrade/blob/eb6c7f8595e826ce7d9ff6413b5b6352c9a5382a/freqtrade/freqtradebot.py#L931-L935

Closes #2785

## Quick changelog

- Add test recreating the faulty behaviour
- Fix bug

### CODE REVIEW

**Comment by hroff-1902** [comment]
LGTM 👍 Great catch, thanks to our bug reporter!
